install/class.Setup.php

```diff
@@ -282,6 +286,52 @@ public function __invoke($params)
         $this->logNotice('Installation completed!');
     }
 
+    /**
+     * @param string $class
+     * @param array  $parametersToSort
+     *
+     * @return array
+     * @throws ReflectionException
+     */
+    private function sortParameters(string $class, array $parametersToSort): array
+    {
+        $reflectionClass = new ReflectionClass($class);
+
+        $constructParameters = $reflectionClass->getMethod('__construct')->getParameters();
+
+        $sortedParameters = [];
+
+        foreach ($constructParameters as $parameter) {
+            $parameterName = $parameter->getName();
+            if (isset($parametersToSort[$parameterName])) {
+                $sortedParameters[$parameter->getPosition()] = $parametersToSort[$parameterName];
+            }
```

Comment:
```suggestion
            } elseif($parameter->isDefaultValueAvailable()) {
                $sortedParameters[$parameter->getPosition()] = $parameter->getDefaultValue();
            } else {
                throw new RuntimeException(
                    sprintf('There is no default parameter for `$%s` in %s::__contruct', $parameter->getName(), $class)
                );
            }
```
and the second loop can be removed I guess, or am I wrong here?

Reply:
the idea of the second loop to cover situation like
`SomeClass::__construct($a, $c='def', $d='def')`

`sortParameters('SomeClass', ['a'=>'val', 'd'=>'val'])`
in this case 'the sorter' adds `c` with default value

Reply:
OK, but in that case `isset($parametersToSort[$parameterName])` will return `false` on a second argument (`$c`) and will add it with a default value to `$sortedParameters[1]`, if you apply my suggestion.

Reply:
I've an even better suggestion. See another comment.

Reply:
I mean, I see your point. But can that case even happen?
If the service was tried to be instantiated with arguments like `'a'=>'val', 'd'=>'val'`, it would've failed anyway.
I'll check deeper tomorrow, but seems to me, the default value setting may be omitted completely then.

Reply:
it won't fail. because of the second loop. it adds default value for the `$c`
this is just an edge case.
i hope it will never happen and all parameters including defaults will be configured

Reply:
No-no, I understand that. I'm just questioning the whole case here. How can it happen that we have values set for argument 1 and 3, but not for 2?

Reply:
And what I'm proposing on top of that is to fill up the set values only.

Reply:
if `2` and `3` have default values and `3` is important for me and `2` is not. so I can skip `2` in seed file and let Setup script fill it with a default value

Reply:
But how can you skip a second argument, and provide a third one?

Reply:
from a seed file
```php
"Service": {
    "type": "configurableService",
    "class": "ServiceClass",
    "options": {
        "first": "value1",
        "third":  "value3"
    }
}
```